modules/PhysiCell_settings.cpp

```diff
@@ -523,6 +523,65 @@ void Parameters<T>::assert_not_exists( std::string search_name )
 	exit(-1);
 }
 
+template <class T>
+void Parameters<T>::update_parameter( std::string my_name , T my_value )
+{
+	// check if variable already exist
+	int parameter_index = -1;
+	auto it = name_to_index_map.find(my_name);
+
+	if (it == name_to_index_map.end()) {
+		std::cout << "Error: parameter named " << my_name << " does not exist. Cannot update the parameter!" << std::endl;
+		exit(-1);
+	}
+
+	// change value
+	parameter_index = it->second;
+	parameters[parameter_index].value = my_value;
+
+	return;
+}
+
+template <class T>
```

Comment:
Can you call the above implementation of `update_parameter` to simplify this function and the codebase?

Reply:
@drbergman could you explain to me how this is done?
Maybe because I am not too familiar with templates.
Could you please give me an example of what you wish for?
Then I will make the changes accordingly.

Reply:
@drbergman  im i not sure if i really can do what you ask for, as  i try to update parameters.
i don't have "default values",  as in the add case.
this is as concise as i can do it.
please write here how i really should do it so that it is acceptable for you. thank you.

![image](https://github.com/user-attachments/assets/3b509590-044f-4291-93df-48a7cf25821f)

Reply:
My intention is something super simple and it seems you took a big step towards it having all 3 of these call `assert_exists`. But something like this could make it even tighter:

```
void Parameters<T>::update_parameter( std::string my_name , T my_value , std::string my_units )
{
	update_parameters( my_name, my_value); // will check my_name exists and then set the value to my_value
	parameters[find_index(my_name)].units = my_units;
	return;
}
```

Reply:
or I think you could even use `parameters[my_name]` instead of `parameters[find_index(my_name)]`

Reply:
also, looking at the `[]` operator on strings, it already checks for existence.

Reply:
i can do that, although i think this will make the code slower.

Reply:
oh yeah? I mean, we could throw in some `const` and pass by reference to reduce any overhead costs. I imagine these functions won't be called nearly enough though to make a noticeable impact on performance. Or will they?

Finally, I personally find this kind of coding to be so much easier to maintain that it easily makes it worth the computational tradeoff (at least as I imagine the magnitude of the cost)

Reply:
is this ok?
![image](https://github.com/user-attachments/assets/d60a9740-e12f-47ee-8a4c-1070577b7c72)

Reply:
hmmm...I don't think I'm able to communicate this...your original is the best of these, in my opinion

Reply:
oh yes, it will get called often. I my case every iteration.
but i know, the PhysiCell core code is not written for only one project. so this argument has no weight.

Reply:
@drbergman could you please confirm that the conflict / conversation / change you requested has resolved (as i think so ).
or let me know if otherwise?

i don't think it is appropriate when i click the resolved button.

thank you, Elmar